research/context/preselection/preselect_wikipedia_comments.py

```diff
@@ -0,0 +1,125 @@
+tf.app.flags.DEFINE_string("comment_type", "ADDITION", "The type of comment ('ADDITION', 'DELETION', 'CREATION', 'MODIFICATION').")
+tf.app.flags.DEFINE_integer("min_txt_len", 1, "Minimum length for comments of the sample.")
+tf.app.flags.DEFINE_integer("max_txt_len", 1000, "Maximum length for comments of the sample.")
+tf.app.flags.DEFINE_string("gbq_table_name", None, "The name of the Google BigQuery table.")
+tf.app.flags.DEFINE_string("gbq_project_name", None, "The name of the Google BigQuery project.")
+tf.app.flags.DEFINE_integer("zones_num", 10, "Number of zones to be used for sampling.")
+tf.app.flags.DEFINE_string("output_csv_path", None, "Path where the output sample CSV should be written.")
+tf.app.flags.DEFINE_string("output_json_path", None, "Path where the output sample JSON should be written.")
+
+def connect_to_gbq(gbq_project_name):
+    """
+    Connect to Google BigQuery
+    :param gbq_project_name: project name
+    :return:
+    """
+    client = bigquery.Client(project=gbq_project_name)
+    return client
+
+def query(low_tox, high_tox, limit, max_txt_len, min_txt_len, type, gbq_table_name):
+    """
+    Google Big Query sampling data of specific types
+    :param low_tox: comments should be higher than that
+    :param high_tox: comments should be lower than that
+    :param limit: fetch some
+    :param type: kind of comments (wikipedia specific)
+    :param max_txt_len: max length of comment
+    :return: dataframe with results
+    """
+    assert low_tox>=0 and high_tox<=1
+    query = """
+    SELECT *, rand() as r 
+        FROM `{gbq_table_name}`
+        WHERE length(cleaned_content)>1 AND length(cleaned_content)<{max_txt_len} AND replyTo_id!="null" AND type="{typ}" AND toxicity>{low} AND toxicity<{high}
+        order by r
+        LIMIT {lim}
+    """.format(gbq_table_name=gbq_table_name, low=low_tox, high=high_tox, lim=limit, typ=type, max_txt_len=max_txt_len, min_txt_len=min_txt_len)
+    query_job = client.query(query)
+    results = query_job.result()
+    return results.to_dataframe()
+
+
+def sample(gbq_table_name, toxic_zones_num, min_txt_len, max_txt_len, type, limit, randomly_selected_limit):
+    """
+    Create a sample of comments using toxic zones and a random batch.
+    :param toxic_zones_num:
+    :return: toxic_zones_num samples + one with randomly selected comments
+    """
+    # Create random sample (initialization)
+    table = query(low_tox=0, high_tox=1, min_txt_len=min_txt_len, max_txt_len=max_txt_len, limit=randomly_selected_limit, type=type, gbq_table_name=gbq_table_name)
+    table["toxic_zone"] = [-1] * table.shape[0]
+    # Add samples following a toxicity-preselection
+    for i in tqdm(range(toxic_zones_num)):
+        preselected = query(low_tox=float(i)/toxic_zones_num, high_tox=float(i+1)/toxic_zones_num, min_txt_len=min_txt_len, max_txt_len=max_txt_len, limit=limit, type=type, gbq_table_name=gbq_table_name)
```

Comment:
This is optional, because it may be a little overkill here, but it's actually much better to combine the queries than run 10 queries. This lets SQL do clever optimizations and gives you only a single big query (see UNION op: https://cloud.google.com/bigquery/docs/reference/standard-sql/query-syntax#union).